### DISCUSSION TITLE
Seeking Guidance on Generating and Including 'fipsmodule.cnf' in the OpenSSL Configuration

### DISCUSSION THREAD

**Discussion by aniljadaun** [openssl/openssl#21967] [Q&A] [answered]
"We plan to include openssl.cnf, fips.dll, and a statically built openssl.exe in our app package, which will be written to the installation folder. We will generate the fipsmodule.cnf file upon the initial launch of our application.

However, our challenge lies in determining the path for generating fipsmodule.cnf in advance, as we do not know the installation directory beforehand. Including fipsmodule.cnf within openssl.cnf is also not feasible. Dumping it into the 'AppData' folder would require user-specific information, making it impractical.

We seek recommendations from the OpenSSL team regarding the optimal location for generating fipsmodule.cnf, ensuring that we can include the path for fipsmodule.cnf within openssl.cnf. How should we approach this?"

**Comment by aniljadaun** [comment]
Should we instead generate openssl.cnf on user machine only  so that we can know and write fipsmodule.cnf path in openssl.cnf? i am confused on this. please suggest.

**Reply by t8m** [reply, marked as answer]
Yes, I think generating the openssl.cnf on the user machine before running the openssl fipsinstall and creating the fipsmodule.cnf is the way to do it.
